app/scripts/controllers/app-state-controller.ts

```diff
@@ -658,6 +664,21 @@ export class AppStateController extends BaseController<
     this.#resetTimer();
   }
 
+  /**
+   * Add account to list of accounts for which user has acknowledged
+   * smart account upgrade splash page.
+   *
+   * @param account
+   */
+  setSplashPageAcknowledgedForAccount(account: string): void {
+    this.update((state) => {
+      state.upgradeSplashPageAcknowledgedForAccounts = [
+        ...state.upgradeSplashPageAcknowledgedForAccounts,
+        account,
```

Comment:
Minor, should we protect against duplicates here?

Reply:
Not very important I think, as user will not see page again if it's added once.